libjcat/jcat-libcrypto-pkcs7-engine.c

```diff
@@ -271,15 +264,11 @@ jcat_libcrypto_pkcs7_engine_verify(JcatEngine *engine,
 			}
 
 			issuer_size = BIO_get_mem_data(issuer_bio, &issuer_string);
-			g_strndup(issuer_string, issuer_size);
-
 			if (issuer_size > 0 && issuer_string != NULL) {
-				/* issuer name isn't null terminated */
 				g_string_overwrite_len(authority_newest,
 						       0,
 						       issuer_string,
 						       issuer_size);
-				g_string_truncate(authority_newest, issuer_size);
```

Comment:
Leaving the `g_strndup` there was an oversight :sweat_smile:.
`g_string_overwrite_len` only writes a `NULL` to terminate a string if it has been lengthened. The `g_string_truncate` call was to terminate the string.
If the loop runs setting the issuer name, then again setting a shorter name the end of the first name will be there in the output.
I've rewritten it to be less complicated.